### PR TITLE
Stop enabling CONFIG_PPC_DISABLE_WERROR=y for clang-14+

### PR DESCRIPTION
**Pull request by nathanchance**
Split up the ppc64 configuration into a "no `-Werror`" variant so that `-Werror` can be enabled for clang-14+. `-Warray-bounds` is only enabled for 5.18+, so we only need this configuration enabled for clang-13 and older on 6.1 and newer.

Closes: https://github.com/ClangBuiltLinux/continuous-integration2/issues/617

### CODE REVIEW

File: generator.yml [hunk 1]
```diff
@@ -311,8 +311,9 @@ configs:
   - &mips              {config: [malta_defconfig, CONFIG_BLK_DEV_INITRD=y, CONFIG_CPU_BIG_ENDIAN=y],           kernel_image: vmlinux,      ARCH: *mips-arch,       << : *kernel}
   - &mipsel            {config: [malta_defconfig, CONFIG_BLK_DEV_INITRD=y],                                    kernel_image: vmlinux,      ARCH: *mips-arch,       << : *kernel}
   - &ppc32             {config: ppc44x_defconfig,                                                              kernel_image: uImage,       ARCH: *powerpc-arch,    << : *kernel}
-  # Disable -Werror for ppc64_guest_defconfig for now: https://github.com/ClangBuiltLinux/linux/issues/1445
-  - &ppc64             {config: [ppc64_guest_defconfig, CONFIG_PPC_DISABLE_WERROR=y],                          kernel_image: vmlinux,      ARCH: *powerpc-arch,    << : *kernel}
+  - &ppc64             {config: ppc64_guest_defconfig,                                                         kernel_image: vmlinux,      ARCH: *powerpc-arch,    << : *kernel}
+  # Disable -Werror for ppc64_guest_defconfig for older clang versions: https://github.com/ClangBuiltLinux/linux/issues/1445
```
Comment (nickdesaulniers) [inline generator.yml:315]: s/older clang versions/clang-13 and older/

**Review by nickdesaulniers — APPROVED** [review]

**Comment by nathanchance** [comment]
> Does this need to wait on [ClangBuiltLinux/linux#252](https://github.com/ClangBuiltLinux/linux/issues/252) (the revert of the revert being accepted)?

Maybe? The build is already broken for `powernv_defconfig` though: https://github.com/ClangBuiltLinux/continuous-integration2/actions/runs/5999812721/job/16275680076

**Comment by nickdesaulniers** [comment]
> > Does this need to wait on [ClangBuiltLinux/linux#252](https://github.com/ClangBuiltLinux/linux/issues/252) (the revert of the revert being accepted)?
> 
> Maybe? The build is already broken for `powernv_defconfig` though: https://github.com/ClangBuiltLinux/continuous-integration2/actions/runs/5999812721/job/16275680076

ah, then I guess not. Merge away!

File: generator.yml (hunk 1 above)
Comment (nathanchance) [reply generator.yml]: Done.

**Pull request merged**